From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Sofiane H. Djerbi" <46628754+kugge@users.noreply.github.com>
Date: Sun, 9 Apr 2023 17:06:46 +0300
Subject: [PATCH] Server mod name


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index f0bd3753d3984d83c49d49e159dc7fc6ffb48bfd..e746cf96a02c7b15bf448e77b297f85d61743d97 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -202,6 +202,9 @@ public class KaiijuConfig {
     private static void optimizationSettings() {
     }
 
+    public static String serverModName = "Kaiiju";
+
     private static void gameplaySettings() {
+        serverModName = getString("gameplay.server-mod-name", serverModName);
     }
 }
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 8c59d02198541e5e37fc12bbb90841a921398519..3bb7b3088295d0195dfa811b360fb5e48c7f9cc5 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -2020,7 +2020,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "Kaiiju"; // Paper // Folia - Folia // Kaiiju - Kaiiju
+        return dev.kaiijumc.kaiiju.KaiijuConfig.serverModName; // Paper // Folia - Folia // Kaiiju - Kaiiju
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
