From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Sofiane H. Djerbi" <46628754+kugge@users.noreply.github.com>
Date: Sat, 8 Apr 2023 23:32:34 +0300
Subject: [PATCH] Gameplay Configuration


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index 94f16629b6c8a8be0df49f72744cf09e2ce94704..f0bd3753d3984d83c49d49e159dc7fc6ffb48bfd 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -201,4 +201,7 @@ public class KaiijuConfig {
 
     private static void optimizationSettings() {
     }
+
+    private static void gameplaySettings() {
+    }
 }
diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
index ffa090cf3f72ae7cf2b6352b2de38e152f032a60..2d5650889bea174605adbc98461a015a474e0801 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
@@ -125,4 +125,7 @@ public class KaiijuWorldConfig {
 
     private void optimizationSettings() {
     }
+
+    private void gameplaySettings() {
+    }
 }
