From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Sofiane H. Djerbi" <46628754+kugge@users.noreply.github.com>
Date: Sun, 7 May 2023 20:04:06 +0300
Subject: [PATCH] Toggle break redstone on top of trap doors early

That patch break vanilla mechanics such as portal slicing.

diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
index 8521a5d266d2337f8ebe2a5d507672236b3e7deb..29e7c6784a6b4eff075975e6409c4566c0f156a2 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuWorldConfig.java
@@ -127,8 +127,10 @@ public class KaiijuWorldConfig {
     }
 
     public boolean fixVoidTrading = true;
+    public boolean breakRedstoneOnTopOfTrapDoorsEarly = true;
 
     private void gameplaySettings() {
         fixVoidTrading = getBoolean("gameplay.fix-void-trading", fixVoidTrading);
+        breakRedstoneOnTopOfTrapDoorsEarly = getBoolean("gameplay.break-redstone-on-top-of-trap-doors-early", breakRedstoneOnTopOfTrapDoorsEarly);
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/world/level/block/TrapDoorBlock.java b/src/main/java/net/minecraft/world/level/block/TrapDoorBlock.java
index 336fcf4af0ffb416b5595a9e65172f36cc36aaa3..f6179f8cc69da924ee0a943968426330e746dc4a 100644
--- a/src/main/java/net/minecraft/world/level/block/TrapDoorBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/TrapDoorBlock.java
@@ -155,6 +155,7 @@ public class TrapDoorBlock extends HorizontalDirectionalBlock implements SimpleW
                 // CraftBukkit end
                 // Paper start - break redstone on trapdoors early
                 boolean open = (Boolean) state.getValue(TrapDoorBlock.OPEN) != flag1;
+                if (world.kaiijuConfig.breakRedstoneOnTopOfTrapDoorsEarly) { // Kaiiju - trigger break redstone on trapdoors early
                 // note: this must run before any state for this block/its neighborus are written to the world
                 // we allow the redstone event to fire so that plugins can block
                 if (flag1 && open) { // if we are now powered and it caused the trap door to open
@@ -171,6 +172,7 @@ public class TrapDoorBlock extends HorizontalDirectionalBlock implements SimpleW
                         }
                     }
                 }
+                } // Kaiiju  - trigger break redstone on trapdoors early
                 if (open) {
                 // Paper end - break redstone on trapdoors early
                     state = (BlockState) state.setValue(TrapDoorBlock.OPEN, flag1);
