From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xymb <xymb@endcrystal.me>
Date: Thu, 22 Jun 2023 00:17:57 +0200
Subject: [PATCH] Option to disable player stats


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index 22a1a2fc8b8fb5fa0cf79f2d5d2c710fd4f71d26..09a69d2c3d1a441c4fc0145116d07508a704e153 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -200,9 +200,11 @@ public class KaiijuConfig {
     }
 
     public static boolean disableVanishApi = false;
+    public static boolean disablePlayerStats = false;
 
     private static void optimizationSettings() {
         disableVanishApi = getBoolean("optimization.disable-vanish-api", disableVanishApi);
+        disablePlayerStats = getBoolean("optimization.disable-player-stats", disablePlayerStats);
     }
 
     private static void gameplaySettings() {
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index e4a68f43c1a48fe8193623c646eb40bc5f664c76..aa6e7e03697023c2c755e0ce119e76e9d708e5d1 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -2414,6 +2414,7 @@ public class ServerPlayer extends net.minecraft.world.entity.player.Player imple
 
     @Override
     public void awardStat(Stat<?> stat, int amount) {
+        if (dev.kaiijumc.kaiiju.KaiijuConfig.disablePlayerStats) return; // Kaiiju - Disable player stats
         this.stats.increment(this, stat, amount);
         this.level().getCraftServer().getScoreboardManager().forAllObjectives(stat, this, (scoreaccess) -> {
             scoreaccess.add(amount);
