From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xymb <xymb@endcrystal.me>
Date: Thu, 22 Jun 2023 00:14:47 +0200
Subject: [PATCH] Option to disable vanish api


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index f0bd3753d3984d83c49d49e159dc7fc6ffb48bfd..22a1a2fc8b8fb5fa0cf79f2d5d2c710fd4f71d26 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -199,7 +199,10 @@ public class KaiijuConfig {
         sendNullEntityPackets = getBoolean("network.send-null-entity-packets", sendNullEntityPackets);
     }
 
+    public static boolean disableVanishApi = false;
+
     private static void optimizationSettings() {
+        disableVanishApi = getBoolean("optimization.disable-vanish-api", disableVanishApi);
     }
 
     private static void gameplaySettings() {
diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index c547d51eebd965f2be7ac45bc0981626c0515ef1..72b0118f349dd2300235bcacfbf12deba7ec367a 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1422,7 +1422,11 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
                 // Folia end - region threading
 
                 // CraftBukkit start - respect vanish API
-                if (flag && (!io.papermc.paper.util.TickThread.isTickThreadFor(player) || !player.getBukkitEntity().canSee(this.entity.getBukkitEntity()))) { // Paper - only consider hits // Folia - region threading
+                // Kaiiju start - don't respect vanish API >:D
+                boolean canSee = true;
+                if (flag && !dev.kaiijumc.kaiiju.KaiijuConfig.disableVanishApi) canSee = player.getBukkitEntity().canSee(this.entity.getBukkitEntity());
+                if (!io.papermc.paper.util.TickThread.isTickThreadFor(player) || !canSee) { // Folia - region threading
+                // Kaiiju end
                     flag = false;
                 }
                 // CraftBukkit end
