From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xymb <xymb@endcrystal.me>
Date: Sat, 20 Jan 2024 16:54:53 +0100
Subject: [PATCH] Optimization: Throttle leaves ticking.


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index b64565bf748c4919e7eeec8f6a294374b78ec010..80b13d4534cb887718a39f06a0dfd5d7c6497adb 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -217,4 +217,13 @@ public class KaiijuConfig {
     private static void highlyExperimentalSettingsDontEnable() {
         disableOWNetherScale = getBoolean("highly-experimental-settings-dont-enable.disable-ow-nether-scale", disableOWNetherScale);
     }
+
+    // Waits for random time of 60s before enabling leaves ticking
+    // Afterwards slows down leaves ticking by 3x
+    // About 8.3% of tick time with 100 bots on a pregenerated world
+    public static boolean throttleLeavesTicking = false;
+
+    private static void optimizationsDirtyWillChangeName() {
+        throttleLeavesTicking = getBoolean("optimizations-dirty-will-change-name.throttle-leaves-ticking", throttleLeavesTicking);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/level/block/LeavesBlock.java b/src/main/java/net/minecraft/world/level/block/LeavesBlock.java
index 07d9696c9bf0a783367a76ebad2273a2c495bc8f..6d570471c059ac0a9f30149778c2ec29baccb8f6 100644
--- a/src/main/java/net/minecraft/world/level/block/LeavesBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/LeavesBlock.java
@@ -76,8 +76,21 @@ public class LeavesBlock extends Block implements SimpleWaterloggedBlock {
         return !(Boolean) state.getValue(LeavesBlock.PERSISTENT) && (Integer) state.getValue(LeavesBlock.DISTANCE) == 7;
     }
 
+    // Kaiiju start
+    int leavesThrottling = -1;
+    // Kaiiju end
+
     @Override
     protected void tick(BlockState state, ServerLevel world, BlockPos pos, RandomSource random) {
+        // Kaiiju start
+        if (dev.kaiijumc.kaiiju.KaiijuConfig.throttleLeavesTicking) {
+            if (leavesThrottling == -1) leavesThrottling = random.nextInt(20 * 60) + 1;
+            leavesThrottling -= 1;
+            if (leavesThrottling >= 0) return;
+            else leavesThrottling = 3;
+        }
+        // Kaiiju end
+
         world.setBlock(pos, LeavesBlock.updateDistance(state, world, pos), 3);
     }
 
