From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xymb <xymb@endcrystal.me>
Date: Thu, 18 Jan 2024 13:34:38 +0100
Subject: [PATCH] Experimental gameplay: Add option to disable OW/Nether 8:1
 scaling.


diff --git a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
index cae27e176af0f4b639875721092cc598d19087ec..5b9d1b184d749f3370c7431bc1aacbe189e8a46f 100644
--- a/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
+++ b/src/main/java/dev/kaiijumc/kaiiju/KaiijuConfig.java
@@ -214,4 +214,10 @@ public class KaiijuConfig {
     private static void gameplaySettings() {
         serverModName = getString("gameplay.server-mod-name", serverModName);
     }
+
+    public static boolean disableOWNetherScale = false; // Changes 8:1 ratio between Nether and OW to 1:1
+
+    private static void highlyExperimentalSettingsDontEnable() {
+        disableOWNetherScale = getBoolean("highly-experimental-settings-dont-enable.disable-ow-nether-scale", disableOWNetherScale);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index d8be7c8dca41b6ad6d9a28590be9e33ef9045bdb..a4b41d98624acab4ecede80f3aa1db41b92efa0b 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -4253,6 +4253,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
 
                 WorldBorder destinationBorder = destination.getWorldBorder();
                 double dimensionScale = DimensionType.getTeleportationScale(origin.dimensionType(), destination.dimensionType());
+                if (dev.kaiijumc.kaiiju.KaiijuConfig.disableOWNetherScale) dimensionScale = 1; // Kaiiju
                 BlockPos targetPos = destination.getWorldBorder().clampToBounds(this.getX() * dimensionScale, this.getY(), this.getZ() * dimensionScale);
 
                 ca.spottedleaf.concurrentutil.completable.Completable<BlockUtil.FoundRectangle> portalFound
